infrastructure-scripts/New-SqlDbServiceAccount.ps1

```diff
@@ -0,0 +1,140 @@
+The Enviroment of the New Service account,
+
+.PARAMETER KeyVaultName
+The name of the Keyvault for the Enviroment
+
+
+.EXAMPLE
+
+.\New-SqlDbServiceAccount.ps1  -ServerName ServerName `
+                                   -DataBaseName DataBaseName `
+                                   -SqlUserName SqlUserName `
+                                   -SqlPassword SqlPassword `
+                                   -SqlServiceAccountName SqlServiceAccountName `
+                                   -SQLResourceGroup SQLResourceGroup
+                                   -Enviroment Enviroment
+                                   -KeyVaultName KeyVaultName
+
+#>
+
+[CmdletBinding(DefaultParameterSetName = 'None')]
+param
+(
+    [Parameter(Mandatory = $true)]
+    [String] $ServerName,
+    [String] $AzureFirewallName = "AzureWebAppFirewall",
+    [Parameter(Mandatory = $true)]
+    [String]  $DataBaseName,
+    [Parameter(Mandatory = $true)]
+    [String]$SqlUserName,
+    [Parameter(Mandatory = $true)]
+    [SecureString]  $SqlPassword,
+    [Parameter(Mandatory = $true)]
+    [String]  $SqlServiceAccountName,
+    [Parameter(Mandatory = $true)]
+    [String]  $SQLResourceGroup,
+    [Parameter(Mandatory = $true)]
+    [String]  $Enviroment,
+    [Parameter(Mandatory = $true)]
+    [String] $KeyVaultName
+)
+
+$ErrorActionPreference = 'Stop'
+
+
+function New-AzureSQLServerFirewallRule {
+    $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+    New-AzureRmSqlServerFirewallRule -StartIPAddress $AgentIp -EndIPAddress $AgentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+function Update-AzureSQLServerFirewallRule {
+    $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+    Set-AzureRmSqlServerFirewallRule -StartIPAddress $AgentIp -EndIPAddress $AgentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+
+function Get-RandomPassword {
+    $Password = ([char[]]([char]33..[char]95) + ([char[]]([char]97..[char]126)) + 0..9 | Sort-Object { Get-Random })[0..33] -join ''
+    return $Password
+}
+function Invoke-SqlQuery([String] $Query) {
+    $Output = Invoke-Sqlcmd -ServerInstance $ServerFQDN -Database $DataBaseName -Username $SqlUserName -Password $SqlPassword -OutputSqlErrors $true -Query $Query
+    return $Output
+}
+
+function Remove-AzureSQLServerFirewallRule {
+    If ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue)) {
+        Remove-AzureRmSqlServerFirewallRule -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+    }
+}
+
+try {
+    $AccountExistQuery = @"
+                        SELECT * FROM sys.database_principals WHERE name = '$SqlServiceAccountName'
+"@
```

Comment:
Does this require a here-string for one line?

Reply:
Yes as if you don't you need excessive amount of back ticking